### ISSUE TITLE
puppet-lint fails to warn on broken regexp

### ISSUE THREAD

**Issue by Katana-Steel** [rodjek/puppet-lint#823]
this doesn't generate a warning/error
```
node /^.+[0-9[a-z]{3}[0-9]+\./ {
  include common
}
```
I think it should...
could possibly also verify validity to `if $x =~ //` tests
